libraries/aws_kms_key.rb

```diff
@@ -15,8 +15,11 @@ def initialize(opts = {})
     # SDK permits key_id to hold either an ID or an ARN
     opts = { key_id: opts } if opts.is_a?(String)
     super(opts)
-    validate_parameters(required: [:key_id])
-
+    if opts[:alias]
+      @alias = opts[:alias]
+      opts[:key_id] = fetch_key_id
```

Comment:
I think we should only perform the "get by alias" if `!opts.is_a?(String) && !opts[:key_id]`. That is to say we should always prefer Key ID over alias. If a Key ID and Alias are both passed - use the Key ID. Only get by Alias if it is the only thing provided (by means of `opts[:alias]`.

So this line would become `if opts[:alias] && !opts[:key_id]`.